src/wp-includes/blocks.php

```diff
@@ -221,9 +222,21 @@ function register_block_style_handle( $metadata, $field_name, $index = 0 ) {
 	if ( $is_core_block ) {
 		$style_path = "style$suffix.css";
 	}
+	$style_path_raw = dirname( $metadata['file'] ) . '/' . $style_path;
 
-	$style_path_norm = wp_normalize_path( realpath( dirname( $metadata['file'] ) . '/' . $style_path ) );
-	$has_style_file  = '' !== $style_path_norm;
+	static $styles_path_norm = array();
+	if ( ! $styles_path_norm ) {
+		$styles_path_norm = array();
+	}
+
+	if ( isset( $styles_path_norm[ $style_path_raw ] ) ) {
+		$style_path_norm = $styles_path_norm[ $style_path_raw ];
+	} else {
+		$style_path_norm                     = $lookup && file_exists( $style_path ) ? wp_normalize_path( realpath( $style_path ) ) : '';
+		$styles_path_norm[ $style_path_raw ] = $style_path_norm;
+	}
```

Comment:
@joemcgill @felixarntz While working on this, I discoveried that the same css can be regsiter for style and editor style. Resulting the same lookup for the same file. This is wasteful, adding this in memory cache, halfs the number of realpath lookups.